### PR TITLE
Move tests from parser.rs to appropriate parse_XX tests

### PR DESCRIPTION
**Pull request by alamb**
Follow on to @ankrgyl 's suggestion in https://github.com/sqlparser-rs/sqlparser-rs/pull/842/files#r1149575835

I tried to clean up the tests a little both to make the current code a little nicer but also to help future contributors find the right place (and the right pattern)

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -6912,40 +6912,6 @@ mod tests {
         });
     }
 
-    #[test]
-    fn test_parse_limit() {
```
Comment (alamb) [inline src/parser.rs:6916]: I moved this code to sqlparser_common and sqlparser_mysql as appropriate, and used more concise functions to test these dialects

File: src/parser.rs [hunk 2]
```diff
@@ -7402,24 +7368,6 @@ mod tests {
         );
     }
 
-    #[test]
```
Comment (alamb) [inline src/parser.rs:7405]: this is postgres specific so moved to sqlparser_postgres

File: src/parser.rs [hunk 3]
```diff
@@ -7458,11 +7406,4 @@ mod tests {
             ))
         );
     }
-
-    #[test]
-    fn test_update_in_with_subquery() {
```
Comment (alamb) [inline src/parser.rs:7463]: moved to sqlparser_postgres

File: tests/sqlparser_common.rs [hunk 4]
```diff
@@ -1559,65 +1564,6 @@ fn parse_select_group_by() {
     );
 }
 
-#[test]
-fn parse_select_group_by_grouping_sets() {
```
Comment (alamb) [inline tests/sqlparser_common.rs:1563]: these tests are postgres specific (test with generic dialect and postgres) so moved to sqlparser_postgres

File: src/parser.rs [hunk 5]
```diff
@@ -6912,40 +6912,6 @@ mod tests {
...
 
-    #[test]
-    fn test_parse_limit() {
-        let sql = "SELECT * FROM user LIMIT 1";
-        all_dialects().run_parser_method(sql, |parser| {
-            let ast = parser.parse_query().unwrap();
-            assert_eq!(ast.to_string(), sql.to_string());
-        });
-
-        let sql = "SELECT * FROM user LIMIT $1 OFFSET $2";
```
Comment (ankrgyl) [inline src/parser.rs:6923]: Did we lose these tests? I see a mysql dialect test for offset but not one that touches the other dialects. Apologies if I missed it!

**Review by ankrgyl — COMMENTED** [review]
lgtm pending one question!

File: src/parser.rs (hunk 5 above)
Comment (alamb) [reply src/parser.rs:6923]: Good spot. I should have called this out explicitly -- this test was replicated and therefore redundant. 

The pre-existing test in sqlparser_common.rs is almost identical: 

https://github.com/sqlparser-rs/sqlparser-rs/blob/784a19138f269dab4ea3e1ad3c9da565d20c9155/tests/sqlparser_common.rs#L6092-L6104

**Comment by alamb** [comment]
Thanks for the review @ankrgyl

**Pull request merged**